File: Module/Home/src/main/java/com/module/home/game/viewholder/FuncationAreaViewHolder.java
package com.module.home.game.viewholder;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.common.view.DebounceViewClickListener;
import com.common.view.ex.ExImageView;
import com.module.home.R;
import com.module.home.game.adapter.GameAdapter;
import com.module.home.game.model.FuncationModel;

public class FuncationAreaViewHolder extends RecyclerView.ViewHolder {

    ExImageView mTaskIv;
    ExImageView mTaskRedIv;
    ExImageView mRankIv;
    ExImageView mPracticeIv;

    public FuncationAreaViewHolder(View itemView, GameAdapter.GameAdapterListener listener) {
        super(itemView);

        mTaskIv = (ExImageView) itemView.findViewById(R.id.task_iv);
        mTaskRedIv = (ExImageView) itemView.findViewById(R.id.task_red_iv);
        mRankIv = (ExImageView) itemView.findViewById(R.id.rank_iv);
        mPracticeIv = (ExImageView) itemView.findViewById(R.id.practice_iv);

        mTaskIv.setOnClickListener(new DebounceViewClickListener() {
            @Override
            public void clickValid(View v) {
                if (listener != null) {
                    listener.clickTask();
                }
            }
        });

        mRankIv.setOnClickListener(new DebounceViewClickListener() {
            @Override
            public void clickValid(View v) {
                if (listener != null) {
                    listener.clickRank();
                }
            }
        });

        mPracticeIv.setOnClickListener(new DebounceViewClickListener() {
            @Override
            public void clickValid(View v) {
                if (listener != null) {
                    listener.clickPractice();
                }
            }
        });

    }

    public void bindData(FuncationModel model) {
        if (model.isTaskHasRed()) {
            mTaskRedIv.setVisibility(View.VISIBLE);
        } else {
            mTaskRedIv.setVisibility(View.GONE);
        }
    }
}

File: Module/Home/src/main/java/com/module/home/game/viewholder/FuncationAreaViewHolder.kt
package com.module.home.game.viewholder

import android.support.v7.widget.RecyclerView
import android.view.View
import com.common.core.view.setDebounceViewClickListener

import com.common.view.ex.ExImageView
import com.module.home.R
import com.module.home.game.adapter.ClickGameListener

class FuncationAreaViewHolder(itemView: View,
                              listener: ClickGameListener) : RecyclerView.ViewHolder(itemView) {
    private val mTaskIv: ExImageView = itemView.findViewById(R.id.task_iv)
    private val mTaskRedIv: ExImageView = itemView.findViewById(R.id.task_red_iv)
    private val mRankIv: ExImageView = itemView.findViewById(R.id.rank_iv)
    private val mPracticeIv: ExImageView = itemView.findViewById(R.id.practice_iv)
    private val mMallIv: ExImageView = itemView.findViewById(R.id.mall_iv)


    init {
        mTaskIv.setDebounceViewClickListener { listener.onClickTaskListener() }
        mRankIv.setDebounceViewClickListener { listener.onClickRankListener() }
        mPracticeIv.setDebounceViewClickListener { listener.onClickPracticeListener() }
        mMallIv.setDebounceViewClickListener { listener.onClickMallListner() }
    }

    fun bindData(isTaskHasRed: Boolean) {
        if (isTaskHasRed) {
            mTaskRedIv.visibility = View.VISIBLE
        } else {
            mTaskRedIv.visibility = View.GONE
        }
    }
}
